gems/smithy/spec/support/client_helper.rb

```diff
@@ -0,0 +1,182 @@
+            'byte' => { 'target' => 'smithy.api#Byte' },
+            'document' => { 'target' => 'smithy.api#Document' },
+            'double' => { 'target' => 'smithy.api#Double' },
+            'enum' => { 'target' => 'smithy.ruby.tests#Enum' },
+            'float' => { 'target' => 'smithy.api#Float' },
+            'intEnum' => { 'target' => 'smithy.ruby.tests#intEnum' },
+            'integer' => { 'target' => 'smithy.api#Integer' },
+            'list' => { 'target' => 'smithy.ruby.tests#List' },
+            'long' => { 'target' => 'smithy.api#Long' },
+            'map' => { 'target' => 'smithy.ruby.tests#Map' },
+            'short' => { 'target' => 'smithy.api#Short' },
+            'string' => { 'target' => 'smithy.api#String' },
+            'structure' => { 'target' => 'smithy.ruby.tests#Structure' },
+            'timestamp' => { 'target' => 'smithy.api#Timestamp' },
+            'union' => { 'target' => 'smithy.ruby.tests#Union' }
+          }
+        },
+        'smithy.ruby.tests#Union' => {
+          'type' => 'union',
+          'members' => {
+            'string' => { 'target' => 'smithy.api#String' }
+          }
+        }
+      }
+    end
+
+    # @param [Array<String>] module_names A list of module names for the
+    #  generated code. For example, `['Company', 'Weather']` would generate
+    #  code in the `Company::Weather` namespace.
+    # @param [Symbol] type The type of artifact to generate. For example,
+    #  `:schema` or `:client`.
+    # @param [Hash] options Additional options to pass to the generator.
+    # (See Plan#initialize)
+    # @option options [String] :fixture The name of the fixture to load in the
+    #  fixtures folder relative to this file.
+    # @option options [Hash] :model The model to generate code for. Defaults to
+    #  a sample model.
+    # @option options [String] :smithy The smithy version. Defaults to '2.0'.
+    # @option options [Hash] :shapes The shapes to generate code for. Defaults to
+    #  a sample set of shapes.
+    # def sample(module_names, type, options = {})
+    #   model = load_model(options)
+    #   plan = create_plan(module_names, model, type, options)
+    #   if options[:fixture]
+    #     sourced_client(plan)
+    #   else
+    #     generated_client(plan)
+    #   end
+    # end
+
+    def generate(type, options = {})
+      model = load_model(options)
+      options[:destination_root] ||= Dir.mktmpdir
+      plan = create_plan(model, type, options)
+      Smithy.generate(plan)
+      plan
+    end
+
+    def source(type, options = {})
+      model = load_model(options)
+      # options[:module_name] ||= next_sample_module_name
+      plan = create_plan(model, type, options)
+      source = Smithy.source(plan)
+      [plan.module_name, source]
+    end
+
+    def cleanup_gem(module_name, tmpdir = nil)
+      undefine_module(module_name)
+      return unless tmpdir
+
+      if ENV.fetch('SMITHY_RUBY_KEEP_GENERATED_SOURCE', 'false') == 'true'
```

Comment:
More general comment on how we're checking ENV flags.  Here we are expecting a case sensitive "true" only.  In my RBS rpy setup I only check if its set at all.   These are only used for our generation and not for customers configuring an SDK so its probably less critical, but we should probably be consistent.

Reply:
I am preferring downcasing. For customer experience we would probably force it downcase.